assets/src/edit-story/components/library/panes/media/deleteDialog.js

```diff
@@ -0,0 +1,132 @@
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../../../app/api';
+import { Plain } from '../../../../components/button';
+import Dialog from '../../../../components/dialog';
+import { useSnackbar } from '../../../../app/snackbar';
+import { useMedia } from '../../../../app/media';
+
+const DialogBody = styled.div`
+  width: 512px;
+  font-family: ${({ theme }) => theme.fonts.body1.family};
+  line-height: ${({ theme }) => theme.fonts.body1.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.body1.weight};
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+const Bold = styled.span`
+  font-weight: bold;
+`;
+
+/**
+ * Display a confirmation dialog for when a user wants to delete a media element.
+ *
+ * @param {Object} props Component props.
+ * @param {number} props.mediaId Selected media element.
+ * @param {boolean} props.showDeleteDialog If dialog should be displayed.
+ * @param {function(boolean)} props.setShowDeleteDialog Callback to toggle dialog display.
+ * @return {null|*} The dialog element.
+ */
+function DeleteDialog({ mediaId, showDeleteDialog, setShowDeleteDialog }) {
+  const {
+    actions: { deleteMedia },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+  const {
+    actions: { deleteMediaElement },
+  } = useMedia();
+
+  const [shouldDelete, setShouldDelete] = useState(false);
+
+  useEffect(() => {
+    const deleteMediaItem = async () => {
+      setShouldDelete(false);
+      setShowDeleteDialog(false);
+      try {
+        await deleteMedia(mediaId);
+        deleteMediaElement({ id: mediaId });
+      } catch (err) {
+        showSnackbar({
+          message: __('Failed to delete media item.', 'web-stories'),
+        });
+      }
+    };
+    shouldDelete ? deleteMediaItem() : null;
+  }, [
+    deleteMedia,
+    deleteMediaElement,
+    mediaId,
+    setShowDeleteDialog,
+    shouldDelete,
+    showSnackbar,
+  ]);
+
+  // Keep icon and menu displayed if menu is open (even if user's mouse leaves the area).
+  return (
+    <Dialog
+      open={showDeleteDialog}
+      onClose={() => setShowDeleteDialog(false)}
+      title={__('Delete image/video?', 'web-stories')}
+      actions={
+        <>
+          <Plain
+            data-testid="cancel"
+            onClick={() => setShowDeleteDialog(false)}
+          >
+            {__('Cancel', 'web-stories')}
+          </Plain>
+          <Plain data-testid="delete" onClick={() => setShouldDelete(true)}>
+            {__('Delete', 'web-stories')}
+          </Plain>
+        </>
+      }
+    >
+      <DialogBody>
+        {__(
+          'You are about to permanently delete this image/video from your site. ' +
+            'The image/video will appear broken in any stories that uses it. ',
```

Comment:
Not just stories. also all other content pieces across the entire WP installation.

Reply:
I got this from the UX mocks, how about "any Wordpress content that uses it" ?

Reply:
Unresolving as it hasn't been adapted yet?

If not changing to "any WordPress content that uses it" (note the capital P), it should at least be "in any stories that **use** it." (note the missing `s`)

Reply:
How about using `sprintf` and doing something like this. 
```
You are about to permanently delete this %s from your site. Any references to this %s through your site will be broken
```

Reply:
Extracting them out to separate variables like in https://github.com/google/web-stories-wp/pull/1733/files/9c53338bb04e08fc00626f466969d30bdba87956#diff-f23faa359a2b930fadcc47d113718db5R101